provisioner/aws-state.js

```diff
@@ -0,0 +1,292 @@
+'use strict';
+
+var Promise = require('promise');
+var lodash = require('lodash');
+var util = require('util');
+var Cache = require('../cache');
+var assert = require('assert');
+var debug = require('debug')('aws-provisioner:aws-state');
+
+
+/**
+ * Return a promise that resolves to an AWS State object.  We do this
+ * so that we can grab the state using promises but then inspect state
+ * using synchronus methods, as they're all just list processing
+ */
+function fetchState(ec2, keyPrefix) {
```

Comment:
Why not just export the constructor for the `AwsState` object, and the put an `update` method on the `AwsState` object.

Then you validate config input in the constructor, set things like `ec2` and `keyPrefix` as properties on `this` and share them between methods that needs them.